src/in-depth/machine-communication.md

```diff
@@ -230,10 +230,38 @@ This is how Visual Studio Code uses _ripgrep_ for its code search.
 
 ## How to deal with input piped into us
 
-<aside class="todo">
 
-**TODO:**
-Talk about how work with stdin
-(see [#95](https://github.com/rust-cli/meta/issues/95))
+Rust programs can read data passed in via `stdin`, whether that's through
+piping or text entered while the program is running. Rust's standard library
+provides us with
+[`std::io::stdin`](https://doc.rust-lang.org/std/io/fn.stdin.html) that can
+read the lines passed in.
 
-</aside>
+Here's a program that counts the words in all of the lines of what's passed in
+via `stdin`. Note, the `-` arg for accepting input from stdin comes from the [Command Line Interface Guidelines](https://clig.dev), a helpful resource when building CLIs.
+
+``` rust,ignore
+{{#include machine-communication-stdin.rs}}
+```
+
+If you run that program with text piped in and the `-` arg, it'll output the
+word count:
+
+``` console
+$ echo "hi there friend\!" | cargo run -- -
+Total words from stdin: 3
+```
+
+stdin can also be used interactively. If you run the program with just `cargo
+run -- -`, nothing is displayed but you can enter text. This approach to
+reading stdin is a blocking behavior for you program. If you enter text and
+press Ctrl+D (sends an End of File (EOF) signal to stdin), it'll output the
+word count. Nifty!
+
+``` console
+$ cargo run --
+Hello there.
+My name is Ferris!
+^D
+Total words from stdin: 6
+```
```

Comment:
I can't quite put my finger on it but the content isn't working for me.

I think one aspect is for us to remember that this is in a section "[Communicating with machines](https://rust-cli.github.io/book/in-depth/machine-communication.html#communicating-with-machines)".

Maybe if we start with a program that reads from a file passed in and then we add stdin support, explaining why we are making different changes.

Also, I wouldn't say "its because clig says so".  That is just a collection of lessons people have learned and conventions that have existed long before it.

Reply:
I revised the content of the section to follow your suggestion of starting with the program that reads the file from disk and then makes it support stdin as a follow up. I appreciate the feedback for making it more clear.